### PR TITLE
Instrument jdk

### PR DESCRIPTION
**Pull request by vext01**
Adds instrument flag to the JavaVMDef which when True emits VM events into log file.

OK?

### CODE REVIEW

File: iterations_runners/iterations_runner.java [hunk 1]
```diff
@@ -5,6 +5,97 @@
...
 import java.util.Arrays;
+import java.util.List;
+
+// Instrumentation imports
+import java.lang.management.ManagementFactory;
+import javax.management.MBeanServer;
+import javax.management.ObjectName;
+import java.lang.management.ManagementFactory;
+import java.lang.management.CompilationMXBean;
+import java.lang.management.GarbageCollectorMXBean;
```
Comment (snim2) [inline iterations_runners/iterations_runner.java]: Garbage Collector Bean!!!!!

File: krun/vm_defs.py [hunk 2]
```diff
@@ -422,6 +424,30 @@ def check_benchmark_files(self, benchmark, entry_point):
...
+        iteration). See the comment in the iterations runner for the format of
+        the records."""
+
+        iter_num = 0
+        iter_data = []
+        prefix_len = len(JavaVMDef.INSTR_MARKER)
+        for line in file_handle:
+            if not line.startswith(JavaVMDef.INSTR_MARKER):
+                continue
+
```
Comment (snim2) [inline krun/vm_defs.py]: Blank line not really needed

File: krun/vm_defs.py [hunk 3]
```diff
@@ -318,7 +317,7 @@ def __eq__(self, other):
 
     def parse_instr_stderr_file(self, file_handle):
         """Parse instrumentation data. Override as necessary"""
-        pass
+        return {}
```
Comment (snim2) [inline krun/vm_defs.py]: Do we say `{}` or `dict()` elsewhere in krun?

File: iterations_runners/iterations_runner.java [hunk 4]
```diff
@@ -5,6 +5,97 @@
...
+         * [collectorName: [[PoolName1, ..., PoolNameN],
+         *                  cumuCollectTime, cumuCollectCount]
+         *
+         * Here "cumu" means "cumulative" and times are in milliseconds.
+         *
+         * Each collector manages one or more memory pools, and one memory
+         * pool may be managed by multiple collectors.
+         *
+         */
+
```
Comment (snim2) [inline iterations_runners/iterations_runner.java]: No need for a blank line here, I think. I'm not really sure why the comment is within the block and not a Javadoc above the method definition?

**Comment by snim2** [comment]
Looks OK to me, apart from those minor comments.

File: krun/vm_defs.py (hunk 3 above)
Comment (vext01) [inline krun/vm_defs.py]: I would only use `dict` if I were building a dict from a list of tuples or something.

File: iterations_runners/iterations_runner.java (hunk 1 above)
Comment (vext01) [inline iterations_runners/iterations_runner.java]: Huh? Are you feeling alright? :)

File: krun/vm_defs.py (hunk 3 above)
Comment (snim2) [inline krun/vm_defs.py]: A quick grep shows that we have used both `{}` and `dict()` to construct empty dictionaries. It isn't important for this PR, and it isn't really worth a PR of its own, but we should probably settle on something to use in future.

Comment (vext01) [inline krun/vm_defs.py]: `{}`, as it's shorter please :)

**Comment by vext01** [comment]
Try this...

**Comment by snim2** [comment]
This looks ready for squashing to me...

**Comment by vext01** [comment]
Kerplunk.

**Pull request merged**